### PR TITLE
feat: Add confirmation to Mac Next Steps card

### PR DESCRIPTION
**Pull request by vkraucunas**
**Asana Task/Github Issue:** https://app.asana.com/0/1208403786889309/1208892824152048/f

## Description
- Adds handling for a confirmation state in the NextStepsCard, to handle the MacOS NTP use-case with “addAppToDockMac"

## Testing Steps
- [Visit here](https://deploy-preview-1300--content-scope-scripts.netlify.app/build/pages/new-tab/?next-steps=addAppToDockMac&next-steps=blockCookies)
- Click the action button in the Add App to Dock NextStepsCard. Should switch to text and icon saying “Added App to Dock!"
- Clicking the action button in the Cookies NextStepCard should do nothing

## Checklist

*Please tick all that apply:*

- [ ] I have tested this change locally
- [ ] I have tested this change locally in all supported browsers
- [ ] This change will be visible to users
- [ ] I have added automated tests that cover this change
- [ ] I have ensured the change is gated by config
- [ ] This change was covered by a ship review
- [ ] This change was covered by a tech design
- [ ] Any dependent config has been merged

### CODE REVIEW

File: special-pages/pages/new-tab/app/next-steps/components/NextSteps.module.css [hunk 1]
```diff
@@ -89,6 +107,27 @@
...
+    svg {
+        height: 1rem;
+        width: 1rem;
+        margin-right: var(--sp-2);
+    }
+    
+    p {
+        max-width: 8rem;
+        font-weight: 600;
+        font-size: calc(11 * var(--px-in-rem));
```
Comment (shakyShane) [inline special-pages/pages/new-tab/app/next-steps/components/NextSteps.module.css:127]: This is already set above, and will inherit I believe

File: special-pages/pages/new-tab/app/next-steps/components/NextStepsCard.js [hunk 2]
```diff
@@ -14,14 +18,32 @@ import { useTypedTranslationWith } from '../../types';
 export function NextStepsCard({ type, dismiss, action }) {
     const { t } = useTypedTranslationWith(/** @type {import("../strings.json")} */ ({}));
     const message = variants[type]?.(t);
+    const [showConfirmation, setShowConfirmation] = useState(false);
+    const hasConfirmationState = message.id === 'addAppToDockMac';
```
Comment (shakyShane) [inline special-pages/pages/new-tab/app/next-steps/components/NextStepsCard.js:22]: We  can move this into the `nextsteps.data` to start making them more data-driven in general and avoid this check - add an optional field with the same name (or whatever you think makes sense) and then just check for it here

File: special-pages/pages/new-tab/app/next-steps/integrations-tests/next-steps.spec.js [hunk 3]
```diff
@@ -58,4 +58,14 @@ test.describe('newtab NextSteps cards', () => {
...
+
+    test('shows a confirmation state', async ({ page }, workerInfo) => {
+        const ntp = NewtabPage.create(page, workerInfo);
+        await ntp.reducedMotion();
+        await ntp.openPage({ nextSteps: ['addAppToDockMac', 'defaultApp'] });
+        await page.getByRole('button', { name: 'Add to Dock' }).click();
+
+        await expect(page.getByText('Added to Dock!')).toBeVisible();
+        await expect(page.getByRole('button', { name: 'Add to Dock' })).not.toBeVisible();
+    });
```
Comment (shakyShane) [inline special-pages/pages/new-tab/app/next-steps/integrations-tests/next-steps.spec.js:70]: Nice! thanks for adding this

File: special-pages/pages/new-tab/app/next-steps/components/NextStepsCard.js (hunk 2 above)
Comment (shakyShane) [reply special-pages/pages/new-tab/app/next-steps/components/NextStepsCard.js:22]: You could even just use the presence of 'confirmationText' for now - but either way I can see other cards having additional states, so we should consider ways to encode their alternative states in data rather that in the components :)

**Review by shakyShane — CHANGES_REQUESTED** [review]

File: special-pages/pages/new-tab/app/next-steps/components/NextSteps.module.css (hunk 1 above)
Comment (vkraucunas) [reply special-pages/pages/new-tab/app/next-steps/components/NextSteps.module.css:123]: good catch. moved the text-specific styling down to this p tag section

File: special-pages/pages/new-tab/app/next-steps/components/NextStepsCard.js (hunk 2 above)
Comment (vkraucunas) [reply special-pages/pages/new-tab/app/next-steps/components/NextStepsCard.js]: This is ready for a new look.

**Review by shakyShane — APPROVED** [review]

**Pull request merged**